homeassistant/components/microbees/climate.py

```diff
@@ -0,0 +1,142 @@
+"""Climate integration microBees."""
+
+from typing import Any
+
+from microBeesPy import Sensor
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import MicroBeesUpdateCoordinator
+from .entity import MicroBeesActuatorEntity
+
+CLIMATE_PRODUCT_IDS = {76, 78}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the microBees climate platform."""
+    coordinator: MicroBeesUpdateCoordinator = hass.data[DOMAIN][
+        entry.entry_id
+    ].coordinator
+    async_add_entities(
+        MBClimate(
+            coordinator,
+            bee_id,
+            bee.actuators[0].id,
+            next(
+                sensor.id
+                for sensor in bee.sensors
+                if sensor.deviceID == (762 if bee.productID == 76 else 782)
+            ),
+        )
+        for bee_id, bee in coordinator.data.bees.items()
+        if bee.productID in CLIMATE_PRODUCT_IDS
+    )
+
+
+class MBClimate(MicroBeesActuatorEntity, ClimateEntity):
+    """Representation of a microBees climate."""
+
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_target_temperature_step = 0.5
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE
+        | ClimateEntityFeature.TURN_ON
+        | ClimateEntityFeature.TURN_OFF
+    )
+    _attr_hvac_modes = [HVACMode.HEAT, HVACMode.OFF]
+    _attr_fan_modes = None
+    _attr_min_temp = 15
+    _attr_max_temp = 35
+    _attr_name = None
+
+    def __init__(
+        self,
+        coordinator: MicroBeesUpdateCoordinator,
+        bee_id: int,
+        actuator_id: int,
+        sensor_id: int,
+    ) -> None:
+        """Initialize the microBees climate."""
+        super().__init__(coordinator, bee_id, actuator_id)
+        self.sensor_id = sensor_id
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return the sensor temperature."""
+        return self.temperature_sensor.value
+
+    @property
+    def temperature_sensor(self) -> Sensor:
+        """Return the sensor temperature."""
+        return self.coordinator.data.sensors[self.sensor_id]
```

Comment:
There's no need for the `temperature_sensor` property if it's only used once IMHO
```suggestion
    @property
    def current_temperature(self) -> float | None:
        """Return the sensor temperature."""
        return self.coordinator.data.sensors[self.sensor_id]
```